src/test/server/store-test.js

```diff
@@ -0,0 +1,201 @@
+            it('Should emit userAdded event', () => {
+                users.push(TEST_USER);
+                UserAction.addNewUser(TEST_USER);
+                assert(UserStore.getUsers(), users);
+                assert(UserStore._users[UserStore._users.length - 1].id, UserStore._users.length);
+            });
+
+            it('Should emit addingFailed event with duplicateUserId if user exists ', () => {
+                UserAction.addNewUser(TEST_USER_EXISTING);
+                assert(UserStore.getUsers(), users);
+                assert(userId, 2);
+            });
+
+            it('Should emit userAdded event if user exists, but force flag is true', () => {
+                users.push(TEST_USER_EXISTING);
+                UserAction.addNewUser(TEST_USER_EXISTING, true);
+                assert(UserStore.getUsers(), users);
+            });
+
+            it('Should emit usersFound event with foundUsers and searchFields if we are in search mode', () => {
+                users.push(TEST_USER);
+                UserAction.findUser(SEARCH_USER_INFO);
+                UserAction.addNewUser(TEST_USER);
+                assert(UserStore.getUsers(), users);
+                assert(foundUsers, []);
+                assert(searchFields, SEARCH_FIELDS);
+            });
+        });
+
+        describe('updateUser', () => {
+            it('Should emit change event', () => {
+                assert.notEqual(users[1], UPDATE_USER);
+                users[1] = UPDATE_USER;
+                UserAction.updateUser(UPDATE_USER);
+                assert(UserStore.getUsers(), users);
+            });
+
+            it('Should emit usersFound event with foundUsers and searchFields if update in search mode', () => {
+                UserAction.findUser(SEARCH_USER_INFO);
+                assert(foundUsers.length, 1);
+                assert.notEqual(users[1], UPDATE_USER);
+                UserAction.updateUser(UPDATE_USER);
+                users[1] = UPDATE_USER;
+                assert(foundUsers, []);
+                assert(UserStore.getUsers(), users);
+            });
+        });
+
+        describe('findUser', () => {
+            it('Should emit usersFound event with empty foundUsers and searchFields if users not found', () => {
+                UserAction.findUser(SEARCH_USER_INFO);
+                assert(foundUsers, []);
+                assert(searchFields, SEARCH_FIELDS);
+            });
+
+            it('Should emit usersFound event with foundUsers and searchFields if user found', () => {
+                UserAction.findUser(SEARCH_USER_INFO);
+                assert(foundUsers.length, 1);
+                assert(searchFields, SEARCH_FIELDS);
+            });
+        });
+
+        describe('stopFindUser', () => {
+            it('Should emit change action and clear searchedUser if stop search user', () => {
+                UserAction.stopFindUser();
+                assert(UserStore.getUsers(), users);
```

Comment:
we can skip checking `users` here but we should check that `change` event is emitted